File: RemoveNode.java
public class RemoveNode {
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
// Time Complexity o(n) space complexity o(1)
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode dummy = new ListNode(-1);
        dummy.next = head;
        ListNode fast = dummy;
        ListNode slow = dummy;
        int count = 0;
        while(count<=n){
            fast=fast.next;
            count++;
        }
        while(fast!=null){
            fast=fast.next;
            slow=slow.next;
        }
        ListNode temp = slow.next;
        slow.next=slow.next.next;
        temp.next=null;
        return dummy.next;
        
    }
}

File: Remove Node.py
# Time complexity : O(1)
# Space complexity : O(1)

# The code ran on Leetcode

# Copy data from the next node and update the next pointer to skip the current next node

class Solution:
    #Function to delete a node without any reference to head pointer.
    def deleteNode(self,curr_node):
        #code here
        next_node = curr_node.next
        if next_node == None:
            curr_node = None
        else:
            curr_node.data = next_node.data
            curr_node.next = curr_node.next.next
